storage/src/storage/sqlite/sync.rs

```diff
@@ -0,0 +1,923 @@
+            WHERE parent.hash = ?
+            ORDER BY blocks.hash
+        ",
+        )?;
+        let out = stmt
+            .query_map([hash], |row| row.get(0))?
+            .collect::<rusqlite::Result<Vec<Digest>>>()?;
+        Ok(out)
+    }
+
+    fn get_transaction_location(&mut self, transaction_id: &Digest) -> Result<Option<TransactionLocation>> {
+        self.conn
+            .query_row(
+                r"
+        SELECT
+        transaction_blocks.block_order,
+        blocks.hash
+        FROM transactions
+        INNER JOIN transaction_blocks ON transaction_blocks.tranasction_id = transactions.id
+        INNER JOIN blocks ON blocks.id = transaction_blocks.block_id
+        WHERE transactions.transaction_id = ? AND blocks.canon_height IS NOT NULL
+        ",
+                [transaction_id],
+                |row| {
+                    Ok(TransactionLocation {
+                        index: row.get(0)?,
+                        block_hash: row.get(1)?,
+                    })
+                },
+            )
+            .optional()
+            .map_err(Into::into)
+    }
+
+    fn get_transaction(&mut self, transaction_id: &Digest) -> Result<SerialTransaction> {
+        self.conn
+            .query_row(
+                r"
+        SELECT
+            transactions.transaction_id,
+            network,
+            ledger_digest,
+            old_serial_number1,
+            old_serial_number2,
+            new_commitment1,
+            new_commitment2,
+            program_commitment,
+            local_data_root,
+            value_balance,
+            signature1,
+            signature2,
+            new_record1,
+            new_record2,
+            proof,
+            memo,
+            inner_circuit_id
+        FROM transactions
+        WHERE transactions.transaction_id = ?
+        ",
+                [transaction_id],
+                |row| {
+                    Ok(SerialTransaction {
+                        id: read_static_blob(row, 0)?,
+                        network: Network::from_id(row.get(1)?),
+                        ledger_digest: row.get(2)?,
+                        old_serial_numbers: vec![row.get(3)?, row.get(4)?],
+                        new_commitments: vec![row.get(5)?, row.get(6)?],
+                        program_commitment: row.get(7)?,
+                        local_data_root: row.get(8)?,
+                        value_balance: AleoAmount(row.get(9)?),
+                        signatures: vec![row.get(10)?, row.get(11)?],
+                        new_records: vec![row.get(12)?, row.get(13)?],
+                        transaction_proof: row.get(14)?,
+                        memorandum: row.get(15)?,
+                        inner_circuit_id: row.get(16)?,
+                    })
+                },
+            )
+            .map_err(Into::into)
+    }
+
+    fn store_init_digest(&mut self, digest: &Digest) -> Result<()> {
```

Comment:
I'm not sure how to solve this properly without storing the digests separately; on one hand, we do want the genesis transactions to have the digest available, but on the other, this is a hack; I think we should now instead special-case this digest (in case it's ever queried for) instead of adding a dummy record; this should also simplify things a bit, as this method could then be removed

Reply:
part of the reason for searching for this particular digest was ensuring that the relationship between all the stored values is enforced in the key-value store model, but the relational model no longer requires this sort of validation